### PR TITLE
HPCC-12350 Default string parameters to const char *

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday gavin.halliday@lexisnexis.com

### CODE REVIEW

**Comment by ghalliday** [comment]
@shamser please review

**Comment by richardkchapman** [comment]
Does this need flagging in the RedBook?

File: ecl/hql/hqlutil.cpp [hunk 1]
```diff
@@ -7381,7 +7381,7 @@ class GccCppNameMangler
             ITypeInfo *paramType = param->queryType();
 
             bool isOut = param->hasAttribute(outAtom);
-            bool isConst = param->hasAttribute(constAtom);
+            bool isConst = !param->hasAttribute(noConstAtom);
```
Comment (shamser) [inline ecl/hql/hqlutil.cpp]: Check for constAtom has been replaced with check for noConstAtom.  It looks like constAtom is still valid - isn't it necessary to check for constAtom?

File: ecl/hql/reservedwords.cpp [hunk 2]
```diff
@@ -486,6 +487,7 @@ static const char * eclReserved15[] = { //Actions and statements
     "build",
     "buildindex",
     "checkpoint",
+    "critical",
```
Comment (shamser) [inline ecl/hql/reservedwords.cpp]: This is an unrelated change: critical.

File: ecl/hql/hqlutil.cpp [hunk 3]
```diff
@@ -7629,7 +7629,7 @@ class Vs6CppNameMangler
...
```
Comment (shamser) [inline ecl/hql/hqlutil.cpp]: Is constAtom obsolete now?

**Comment by shamser** [comment]
@ghalliday Should constAtom be eliminated given the potential for confusion in having both constAtom and noConstAtom?

**Comment by ghalliday** [comment]
constAtom is used for other items (e.g., ASSERT), so needs to remain.

const is still allowed in the syntax to ensure backwards compatibility (although it is now ignored)

Yes it is a redbook item if this is taken - it did affect some queries (which had all been incorrectly defined).

**Comment by ghalliday** [comment]
@shamser see above

**Comment by shamser** [comment]
@ghalliday No other comments. Looks good.
